sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -57,6 +64,27 @@ abstract class FileScanBuilder(
     StructType(fields)
   }
 
+  def pushFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Unit = {
+    this.partitionFilters = partitionFilters
+    this.dataFilters = dataFilters
+    this.pushedDataFilters = pushDataFilters(translateDataFilter)
+  }
+
+  protected def pushDataFilters(dataFilters: Array[Filter]): Array[Filter] = Array.empty[Filter]
+
+  def translateDataFilter(): Array[Filter] = {
```

Comment:
can we inline this method in `pushFilters`?

Reply:
can't inline this because it is also used in `ParquetScanFilter`.`pushedParquetFilters`, but I will change it to protected.